### PR TITLE
CCloud-ify dynamic output KT

### PR DESCRIPTION
**Pull request by ybyzek**
### Description

* CCloud-ification of the dynamic output KT
* Note that `ccloud` CLI does not have support for `Long` keys, so this PR modifies the original code (and non-Cloud) KT to use `String` instead of `Long` keys --> source code and test code has changed

### Staging Docs

http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/ccloud-dynamic-output/dynamic-output-topic/confluent.html
http://kafka-tutorials-staging.s3-website-us-west-2.amazonaws.com/ccloud-dynamic-output/dynamic-output-topic/kstreams.html

### CODE REVIEW

File: _includes/tutorials/dynamic-output-topic/kstreams/code/src/main/java/io/confluent/developer/DynamicOutputTopic.java [hunk 1]
```diff
@@ -39,33 +40,22 @@
 
     static final double FAKE_PRICE = 0.467423D;
 
-	public Properties buildStreamsProperties(Properties envProps) {
-        Properties props = new Properties();
-
-        props.put(StreamsConfig.APPLICATION_ID_CONFIG, envProps.getProperty("application.id"));
```
Comment (bbejeck) [inline _includes/tutorials/dynamic-output-topic/kstreams/code/src/main/java/io/confluent/developer/DynamicOutputTopic.java:45]: nice cleanup!

**Review by bbejeck — COMMENTED** [review]
Same as the other tutorial, can you rebase to pick up #789? Other than that, this LGTM.

EDIT - Resolved with comment https://github.com/confluentinc/kafka-tutorials/pull/790#issuecomment-819063312

It would be good to see the test for the dynamic output tutorial pass before merging for a sanity check.

**Comment by bbejeck** [comment]
The test failure was unrelated, I kicked off a new build.

**Comment by ybyzek** [comment]
Kicking off another build in order to get to the second block

**Comment by bbejeck** [comment]
Test for the relevant KT passed https://confluentinc.semaphoreci.com/jobs/31bf5af8-6edb-49ce-b2ad-e95a061117ec

**Review by bbejeck — APPROVED** [review]

**Review by rspurgeon — APPROVED** [review]
LGTM

**Pull request merged**